tmk_core/common/action_layer.h

```diff
@@ -21,24 +21,24 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include "keyboard.h"
 #include "action.h"
 
-
+typedef uint32_t layer_state_t;
```

Comment:
```suggestion
#if defined(LAYER_STATE_8BIT) || defined(DYNAMIC_KEYMAP_ENABLE)
typedef uint8_t layer_state_t;
#elif defined(LAYER_STATE_16BIT)
typedef uint16_t layer_state_t;
#else
typedef uint32_t layer_state_t;
#endif
```

Reply:
The `DYNAMIC_KEYMAP_ENABLE` option is for ... well VIA mostly, which uses EEPROM for keymap store the layers, and is limited to 4 usually.

Reply:
@alex-ong Any thoughts or updates on this?

Reply:
sorry, been busy with work... hopefully i can get these changes done. 
From the TMK pr i remember that although `typedef uint16_t layer_state_t` compiles, its also good to change it in `action.c ` (refer to https://github.com/tmk/tmk_keyboard/pull/526)
Thoughts on that?

Reply:
Not a problem. Work comes first! 

That would be best, definitely.  

However, I think a define is fine for now.   Both the Ergodox EZ and QMK Configurators could check the number of layers and add the define at compile time, until such a time that is possible/added.

Reply:
> ```#if defined(LAYER_STATE_8BIT) || ( defined(DYNAMIC_KEYMAP_ENABLE) && DYNAMIC_KEYMAP_LAYER_COUNT >= 8 )```
Just realised is it meant to be <= 8, or >= 8?